syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package nil.xit.proto;

enum MessageType {
    MessageType_MarkupRequest = 0;
    MessageType_MarkupResponse = 1;
    MessageType_BindingRequest = 2;
    MessageType_BindingResponse = 3;
    MessageType_BindingUpdate = 4;
    MessageType_FileRequest = 5;
    MessageType_FileResponse = 6;
}

message MarkupRequest
{
    string id = 1;
}

message MarkupResponse
{
    string id = 1;
    string file = 2;
}

message Binding
{
    string tag = 1;
    oneof value {
        int64 value_i64 = 2;
        string value_str = 3;
    }
}

message BindingRequest
{
    string id = 1;
}

message BindingResponse
{
    string id = 1;
    repeated Binding bindings = 2;
}

message BindingUpdate
{
    string id = 1;
    Binding binding = 2;
}

message FileRequest
{
    string target = 1;
}

message FileResponse {
    string target = 1;
    string content = 3;
}